docs/specs/2026-08-31-pins-and-bug-class.md

```
@@ -804,3 +813,173 @@ Run the new loop on those pre-fix states; if `classify` groups the class and `re
+  git blob (blobs are whole files), so hashing "the removed text" as a blob and verifying it against
+  `ParentSHA:File` was wrong — a legitimate partial deletion would fail while any file edit
+  (including Guard-and-Go) would pass. The evidence is two separate identities:
+  ```go
+  type DeletionRef struct {
+      File      string // repo-relative path the code was removed from
+      ParentSHA string // a commit where the pre-deletion file still exists (durable in history)
+      FileBlob  string // git blob hash of the WHOLE ParentSHA:File — the durable, content-addressed container
+      Removed   string // the exact removed SPAN (the deleted text) — the deletion's own identity
+  }
+  // idempotent deletion id = hash(File + "\x00" + Removed); FileBlob+ParentSHA anchor it in real history
+  ```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)